network/src/peer_manager/senders.rs

```diff
@@ -100,6 +110,10 @@ impl PeerManagerRequestSender {
             res_tx,
             timeout,
         };
+        fail_point!("network::send::any", |_| {
+            std::thread::sleep(timeout);
```

Comment:
You probably want tokio::time::sleep, no? Otherwise this will block the entire OS thread?

Reply:
it's probably not easy to plug in async function here, but not sure

Reply:
@JoshLind  - don't know how to have async inside of failpoint lambda?

Reply:
Aah, gotcha. Yeah, not sure if failpoints supports that. In which case, ignore me 😄